root/etc/e-smith/events/actions/nethserver-dc-createadmins

```diff
@@ -36,11 +36,7 @@ fi
 /etc/e-smith/events/actions/nethserver-dc-user-lock $EVENT "$ADMIN@$DOMAIN"
 
 # add the admin account to Domain Admins group
-expect <<EOF
-spawn -noecho /usr/bin/systemd-run -M nsdc -q -t /usr/bin/samba-tool group addmembers "Domain Admins" "$ADMIN"
-expect "Added members" { exit 0 }
-exit 3
-EOF
+/usr/bin/nsdc-cmd /usr/bin/samba-tool group addmembers \'Domain Admins\' \'$ADMIN\'
```

Comment:
Quotes escaping is no more required once the nscd-cmd fix is appied

Reply:
Quotes are still needed, even after the fix.

Reply:
Who needs them? It seems quote escaping is required by shell spawned by the dropbear server

```text
[root@vm5 ~]# dbclient  -yy -i /var/lib/nethserver/nsdc/nsdc.key root@192.168.122.55 ls a 'b c'
[...]
[pid  6173] execve("/bin/bash", ["bash", "-c", "ls a b c"], [/* 7 vars */]) = 0
[pid  6173] execve("/usr/bin/ls", ["ls", "a", "b", "c"], [/* 10 vars */]) = 0
```

Compared to

```text
[root@vm5 ~]# dbclient  -yy -i /var/lib/nethserver/nsdc/nsdc.key root@192.168.122.55 ls a \'b c\'
[...]
[pid  6176] execve("/bin/bash", ["bash", "-c", "ls a 'b c'"], [/* 7 vars */]) = 0
[pid  6176] execve("/usr/bin/ls", ["ls", "a", "b c"], [/* 10 vars */]) = 0
```

We must prevent shell injection in general. This issue is not strictly related to dropbear, but we must take it into account, especially where the arguments come from the user's input.